agent/engine/task_manager_linux_test.go

```diff
@@ -16,28 +16,381 @@
+		mockControl.EXPECT().Exists(gomock.Any()).Return(false),
+		mockControl.EXPECT().Create(gomock.Any()).Return(mockCgroup, nil),
+	)
+
+	assert.NoError(t, mtask.setupCgroup())
+}
+
+// TestSetupCgroupInvalidTaskARN attempts to setup a task cgroup for tasks with
+// invalid taskARN's
+func TestSetupCgroupInvalidTaskARN(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mtask := managedTask{
+		Task: &api.Task{
+			Arn: invalidTaskArn,
+		},
+	}
+
+	assert.Error(t, mtask.setupCgroup())
+}
+func TestSetupCgroupInvalidMemConfig(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockControl := mock_cgroup.NewMockControl(ctrl)
+	setControl(mockControl)
+
+	mtask := managedTask{
+		Task: &api.Task{
+			Arn: validTaskArn,
+			Containers: []*api.Container{
+				{
+					Name:   "C1",
+					Memory: uint(1024),
+				},
+			},
+			MemoryLimit: int64(512),
+		},
+	}
+
+	mockControl.EXPECT().Exists(gomock.Any()).Return(false)
+
+	assert.Error(t, mtask.setupCgroup())
+}
+
+// TestSetupCgroupExists validates the path when task cgroup already exists
+func TestSetupCgroupExists(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockControl := mock_cgroup.NewMockControl(ctrl)
+	setControl(mockControl)
+
+	mtask := managedTask{
+		Task: testdata.LoadTask("sleep5TaskCgroup"),
+	}
+
+	mockControl.EXPECT().Exists(gomock.Any()).Return(true)
+
+	err := mtask.setupCgroup()
+	assert.NoError(t, err, "cgroup already exists")
+}
+
+// TestSetupCgroupCreateError validates the path when cgroup creation fails
+func TestSetupCgroupCreateError(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockCgroup := mock_cgroups.NewMockCgroup(ctrl)
+	mockControl := mock_cgroup.NewMockControl(ctrl)
+	setControl(mockControl)
+
+	mtask := managedTask{
+		Task: testdata.LoadTask("sleep5TaskCgroup"),
+	}
+
+	gomock.InOrder(
+		mockControl.EXPECT().Exists(gomock.Any()).Return(false),
+		mockControl.EXPECT().Create(gomock.Any()).Return(mockCgroup, errors.New("cgroup create error")),
+	)
+
+	err := mtask.setupCgroup()
+	assert.Error(t, err, "cgroup create error")
+}
+
+// TestSetupCgroupNil validates when the cgroup object is nil without any error
+func TestSetupCgroupNil(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockControl := mock_cgroup.NewMockControl(ctrl)
+	setControl(mockControl)
+
+	mtask := managedTask{
+		Task: testdata.LoadTask("sleep5TaskCgroup"),
+	}
+
+	gomock.InOrder(
+		mockControl.EXPECT().Exists(gomock.Any()).Return(false),
+		mockControl.EXPECT().Create(gomock.Any()).Return(nil, nil),
+	)
+
+	err := mtask.setupCgroup()
+	assert.Error(t, err)
+}
+
+// TestSetControl validates the set cgroup control method
+func TestSetControl(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockControl := mock_cgroup.NewMockControl(ctrl)
+	setControl(mockControl)
+	assert.Equal(t, mockControl, control)
+}
+
+// Cgroup Cleanup Tests
+
```

Comment:
Please fix these two line, whether remove or rewrite the comments in the format `Testxxxx tests xxx`.